### PR TITLE
Close #1940 Removing unnecessary runtime development-related dependencies

### LINKED ISSUE #1940: Re-evaluate package dependencies to identify/remove those that aren't runtime dependencies
### Background Information

See #1919 where the author of {styler} suggested this course of action for their package. Idea is to test out doing this for {styler} in Q2-2023 release, and then try identify other candidate packages in for Q3 2023 release.

Reasoning (quoting #1919 ):

- Your users may install a package they don't need (assuming most people who install your package are useres, not developers). This costs additional bandwith, installation time, disk space etc... If they need it, they can just manually install it.
- I as a maintainer of {styler} have to run reverse dependency checks (R CMD Check your package with my new version of {styler}) upon submitting to CRAN e.g. with {revdepcheck}. {styler} has more than 40 reverse dependencies, which makes this a long process.

### Definition of Done

Remove package dependencies that we don't need, in order to lessen the load on maintainers of said packages.

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by zdz2101** [comment]
@galachad Similar to how you accomplished https://github.com/pharmaverse/admiral/pull/1942, do you know why only the 4.1 version is failing here for the exclusion or `library(rmarkdown)`?

**Comment by bms63** [comment]
ummm... I would of thought `devtools:document()` was needed to make the admiral-package.Rd file with this update?

**Comment by bms63** [comment]
Hi @cicdguy any ideas on 4.1 failing check for vignettes?

**Comment by zdz2101** [comment]
> ummm... I would of thought `devtools:document()` was needed to make the admiral-package.Rd file with this update?

pretty sure `devtools::document()` is just a wrapper for `roxygen2` which according to the sacred text, ie R Packages (2e) 
image

**Comment by bms63** [comment]
I was thinking that when you update the desciprtion file you need to update the `.Rd` file that is generated from it.  I guess I was mistaken

**Comment by bms63** [comment]
Did we want to try and clean up the dependencies still? @ddsjoberg and @zdz2101 

Perhaps a Wednesday topic and then we can get this PR done?

**Comment by zdz2101** [comment]
> Did we want to try and clean up the dependencies still? @ddsjoberg and @zdz2101
> 
> Perhaps a Wednesday topic and then we can get this PR done?

^agreed, brief wednesday discussion should suffice

**Comment by manciniedoardo** [comment]
Tagging @galachad to look at failing check.

**Comment by galachad** [comment]
Hi @zdz2101, looks like we need the `rmarkdown` package in suggest as it's used for output of vignette: `output: rmarkdown::html_vignette`. 

In my opinion we should keep `rmarkdown` package in `Suggest` as we are using `rmarkdown::html_vignette` and having `rmarkdown` in suggest is default behaviour for all packages. 

I'm not sure why it's only occur on 4.1 but my suspicion is the changes in `devtools` between 2.4.3-2.4.4.

File: DESCRIPTION [hunk 1]
```diff
@@ -56,20 +56,12 @@ Imports:
     tidyselect (>= 1.1.0)
 Suggests:
     admiral.test (>= 0.6.0),
-    covr,
-    devtools,
     DT,
     diffdf,
     knitr,
```
Comment (galachad) [inline DESCRIPTION:61]:
```suggestion
    knitr,
    rmarkdown,
```

**Review by bms63 — COMMENTED** [review]
Do you mind removing magrittr, tidyselect and I think we can drop tibble as well??

**Comment by zdz2101** [comment]
> Do you mind removing magrittr, tidyselect and I think we can drop tibble as well??

This actually brings up a good point of implicit dependencies, if we require `admiraldev` and `admiraldev` lists these all out (how we're getting by with the dplyr >1), should we be redundantly be keeping them here in `admiral` ?

**Comment by bms63** [comment]
> > Do you mind removing magrittr, tidyselect and I think we can drop tibble as well??
> 
> This actually brings up a good point of implicit dependencies, if we require `admiraldev` and `admiraldev` lists these all out (how we're getting by with the dplyr >1), should we be redundantly be keeping them here in `admiral` ?

I think we should keep the main ones for sure.  People check the description file and having them over at admiraldev would be another page to track this stuff down???  @ddsjoberg

**Comment by zdz2101** [comment]
> > > Do you mind removing magrittr, tidyselect and I think we can drop tibble as well??
> > 
> > 
> > This actually brings up a good point of implicit dependencies, if we require `admiraldev` and `admiraldev` lists these all out (how we're getting by with the dplyr >1), should we be redundantly be keeping them here in `admiral` ?
> 
> I think we should keep the main ones for sure. People check the description file and having them over at admiraldev would be another page to track this stuff down??? @ddsjoberg

Nvm, I was given quite an amount of grief playing around as is, I think anything declared in our `R/admiral_package.R` file needs to be included, those 3^ listed above need to be included

**Comment by bundfussr** [comment]
> > > Do you mind removing magrittr, tidyselect and I think we can drop tibble as well??
> > 
> > 
> > This actually brings up a good point of implicit dependencies, if we require `admiraldev` and `admiraldev` lists these all out (how we're getting by with the dplyr >1), should we be redundantly be keeping them here in `admiral` ?
> 
> I think we should keep the main ones for sure. People check the description file and having them over at admiraldev would be another page to track this stuff down??? @ddsjoberg

I would not rely on implicit dependencies. I think if we call a function in admiral code, we should add the package and the required version to `Imports:`. I.e., I would require dplyr > 1 also in admiral because we are calling `across()`.

If we remove `magrittr`, `tidyselect`, and `tibble` from `Imports:`, we would need to update `admiral-package.R` as well and import the functions from `dplyr`. However, I am not sure if we should rely on reexports. Using the original packages looks cleaner to me.

**Comment by ddsjoberg** [comment]
Correct, we cannot rely on implicit dependencies: CRAN would reject us!

We can simple update to re-export functions from dplyr, however: simple and easy, while reducing the number of pkgs listed in our description. I would personally re-export from dplyr, but if you feel strongly against it, either way is fine.

**Comment by bms63** [comment]
Hi all, @manciniedoardo @bundfussr @ddsjoberg @zdz2101 

I think I am losing track of what we were trying to do here.

So if we drop `magrittr` and `tidyselect`  from the `Imports` we will need to update the admiral-pacakge.R as seen below to re-export those functions?

image

**Comment by ddsjoberg** [comment]
The `@importFrom` only imports the function for use _within_ the package. We'd want to also export the functions. This is commonly done in a `reexport.R` script. Here's an example:
https://github.com/pharmaverse/ggsurvfit/blob/main/R/reexport.R

```r
#' @importFrom survival survfit
#' @export
survival::survfit

#' @importFrom survival Surv
#' @export
survival::Surv

#' @importFrom dplyr %>%
#' @export
dplyr::`%>%`
```

**Comment by bundfussr** [comment]
> So if we drop `magrittr` and `tidyselect` from the `Imports` we will need to update the admiral-pacakge.R as seen below to re-export those functions?

It is not related to re-exporting. I think if we remove `magrittr` from `Imports:` but keep `@importFrom magrittr %>%`, R-CMD check will complain.

**Comment by bms63** [comment]
i say lets just leave this issue for after the release as I am still a little confused.

**Review by bms63 — COMMENTED** [review]
@zdz2101 and @ddsjoberg this one is good to go correct?

**Comment by zdz2101** [comment]
> @zdz2101 and @ddsjoberg this one is good to go correct?

For just Suggestions, yes I believe we mentioned we'll make separate issue for Imports

**Review by bms63 — APPROVED** [review]
LGTM!

**Pull request merged**